lib/db/transactions.go

```diff
@@ -83,18 +88,22 @@ func (t readOnlyTransaction) getGlobal(keyBuf, folder, file []byte, truncate boo
 // batch size.
 type readWriteTransaction struct {
 	readOnlyTransaction
-	*batch
+	Batch
 }
 
-func (db *instance) newReadWriteTransaction() readWriteTransaction {
-	return readWriteTransaction{
-		readOnlyTransaction: db.newReadOnlyTransaction(),
-		batch:               db.newBatch(),
+func (db *Lowlevel) newReadWriteTransaction() (readWriteTransaction, error) {
+	t, err := db.newReadOnlyTransaction()
+	if err != nil {
+		return readWriteTransaction{}, err
 	}
+	return readWriteTransaction{
+		readOnlyTransaction: t,
+		Batch:               db.NewBatch(),
+	}, nil
 }
 
 func (t readWriteTransaction) close() {
-	t.flush()
+	_ = t.Flush()
 	t.readOnlyTransaction.close()
```

Comment:
This could return an error, and tell us that it fucked up and didn't do anything so we can let the user know?